lib/connect_client.go

```diff
@@ -325,6 +314,17 @@ func (p connectClient) sync(ctx context.Context, logger DatabaseLogger, tableNam
 			}
 		}
 
+		if res.Cursor != nil {
+			tc = res.Cursor
+		}
+
+		// Because of the ordering of events in a vstream
+		// we receive the vgtid event first and then the rows.
+		// the vgtid event might repeat, but they're ordered.
+		// so we once we reach the desired stop vgtid, we stop the sync session
+		// if we get a newer vgtid.
+		watchForVgGtidChange = watchForVgGtidChange || tc.Position == stopPosition
+
 		if watchForVgGtidChange && tc.Position != stopPosition {
```

Comment:
I think there is a pre-existing bug here. It looks like we are waiting for one event to occur after we've encountered `stopPosition`. Possibly because of the incorrect belief above that `vgtid` events occur before row data, not after.

At some point we had a similar bug in the airbyte connector: https://github.com/planetscale/airbyte-source/pull/131

Not asking you to do anything different in this PR, just making a note of a potential issue.